src/gui/profilepagewidget.cpp

```diff
@@ -0,0 +1,82 @@
+#include "profilepagewidget.h"
+#include "guiutility.h"
+#include "theme.h"
+
+#include <QPushButton>
+
+namespace OCC {
+
+ProfilePageWidget::ProfilePageWidget(QWidget *parent)
+    : QWidget(parent)
+{
+}
+
+void ProfilePageWidget::setProfileConnector(
+    std::unique_ptr<OcsProfileConnector> profileConnector, const QString &userId)
+{
+    _profileConnector = std::move(profileConnector);
+    _profileConnector->fetchHovercard(userId);
+    connect(
+        _profileConnector.get(), &OcsProfileConnector::hovercardFetched, this, &ProfilePageWidget::onHovercardFetched);
+    connect(_profileConnector.get(), &OcsProfileConnector::iconLoaded, this, &ProfilePageWidget::onIconLoaded);
+}
+
+void ProfilePageWidget::resetLayout()
+{
+    _mainLayout = new QVBoxLayout;
+    _mainLayout->setSpacing(0);
+    setLayout(_mainLayout);
+    _profilePageButtonIcons.clear();
+}
+
+void ProfilePageWidget::createLayout()
+{
+    const auto hovercardActions = _profileConnector->hovercard()._actions;
+    _profilePageButtonIcons.reserve(hovercardActions.size());
+    for (const auto &hovercardAction : hovercardActions) {
+        const auto button = new QPushButton;
```

Comment:
why no parent passed to the constructor ?
it looks to me like a memory leak waiting to happen

Reply:
the layout will take over the ownership according to https://doc.qt.io/qt-5/qlayout.html#addWidget

Reply:
@FlexW `const auto`

Reply:
OK